ironfish/src/rpc/routes/eth/sendTransaction.ts

```diff
@@ -0,0 +1,110 @@
+import * as yup from 'yup'
+import { Assert } from '../../../assert'
+import { GLOBAL_IF_ACCOUNT } from '../../../evm'
+import { FullNode } from '../../../node'
+import { legacyTransactionToEvmDescription } from '../../../primitives'
+import { EthUtils } from '../../../utils/eth'
+import { AssertSpending } from '../../../wallet/account/account'
+import { ApiNamespace } from '../namespaces'
+import { routes } from '../router'
+
+export type EthSendTransactionRequest = {
+  from: string
+  to?: string
+  gas?: string
+  gasPrice?: string
+  value?: string
+  data?: string
+  nonce?: string
+}
+
+export type EthSendTransactionResponse = {
+  result: string
+}
+
+export const EthSendTransactionRequestSchema: yup.ObjectSchema<EthSendTransactionRequest> = yup
+  .object({
+    from: yup.string().defined(),
+    to: yup.string().optional(),
+    gas: yup.string().optional(),
+    gasPrice: yup.string().optional(),
+    value: yup.string().optional(),
+    data: yup.string().optional(),
+    nonce: yup.string().optional(),
+  })
+  .defined()
+
+export const EthSendTransactionResponseSchema: yup.ObjectSchema<EthSendTransactionResponse> =
+  yup
+    .object({
+      result: yup.string().defined(),
+    })
+    .defined()
+
+routes.register<typeof EthSendTransactionRequestSchema, EthSendTransactionResponse>(
+  `${ApiNamespace.eth}/sendTransaction`,
+  EthSendTransactionRequestSchema,
+  async (request, node): Promise<void> => {
+    Assert.isInstanceOf(node, FullNode)
+
+    const account = node.wallet.listAccounts().find((a) => a.ethAddress === request.data.from)
+    Assert.isNotUndefined(account, 'Account not found')
+    AssertSpending(account)
+
+    const ethAccount = account.ethAddress
+      ? await node.chain.evm.getAccount(
+          Address.fromString(EthUtils.prefix0x(account.ethAddress)),
+        )
+      : undefined
+
+    const nonce = request.data.nonce
+      ? BigInt(request.data.nonce)
+      : ethAccount?.nonce ?? BigInt(0)
+
+    const gas = request.data.gas ? BigInt(request.data.gas) : 1000000n
+    const gasPrice = request.data.gasPrice ? BigInt(request.data.gasPrice) : 0n
+    const value = request.data.value ? BigInt(request.data.value) : undefined
+
+    const ethTransaction = new LegacyTransaction({
+      nonce: nonce,
+      to: request.data.to,
+      gasLimit: gas,
+      gasPrice: gasPrice,
+      value: value,
+      data: request.data.data,
+    })
+
+    const signed = ethTransaction.sign(Buffer.from(account.spendingKey, 'hex'))
+
+    const evmDescription = legacyTransactionToEvmDescription(signed)
+    const result = await node.chain.evm.simulateTx({ tx: signed })
+    const events = result.events
+    Assert.isNotUndefined(events)
+
+    const raw = await node.wallet.createEvmTransaction({
+      evm: evmDescription,
+      evmEvents: events,
+      account: account,
+    })
+
+    // TODO: This is pretty hacky to figure out which key to post with
+    const unshields = events.filter((e) => e.name === 'unshield')
```

Comment:
Do you think it's less hacky to check if the transaction has any spends instead?

Reply:
We only need to sign with global account because of the mints right? Could be signed with personal key if we use the balancing approach maybe? Thoughts for later

Reply:
Yea checking spends might be slightly less hacky but I think I'd still keep the comment there bc we want to figure out a better way in the future